### ISSUE TITLE
Closing module adds padding to images

### ISSUE THREAD

**Issue by karhohs** [CellProfiler/CellProfiler#2837]
The closing module will add padding or a border around an image the size of the structuring element.

![image](https://user-images.githubusercontent.com/156232/29383484-96a976da-829e-11e7-84fb-59ad63fbbb4b.png)

This is not a display issue, because subsequent operations on the output of Closing treats this border as part of the image. For instance, Gaussian smoothing blurs the border.

![image](https://user-images.githubusercontent.com/156232/29383546-d1126e62-829e-11e7-8412-aa6f7f3322ca.png)

**Comment by 0x00b1** [comment]
Hi, @karhohs. Would you mind sharing the original image?

**Comment by karhohs** [comment]
No problem. 
[issue_2837.zip](https://github.com/CellProfiler/CellProfiler/files/1231972/issue_2837.zip)

**Comment by mcquin** [comment]
@karhohs We use `skimage.measure.closing` as the closing operation. I believe padding the image border with 0-values is the default behavior of `skimage.morphology.closing`.

**Comment by karhohs** [comment]
@mcquin I believe the [skimage.morphology.closing example](http://scikit-image.org/docs/dev/api/skimage.morphology.html#skimage.morphology.closing) does not show this padding.

in:
```
[[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0],
[1, 1, 0, 1, 1],
[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0]]
```

out:
```
[[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0],
[1, 1, 1, 1, 1],
[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0]]
```

**Comment by karhohs** [comment]
Ah ha! The issue is not related to *skimage.morphology.closing*, but instead ***skimage.morphology.binary_closing***.

https://gist.github.com/karhohs/bad263764e0b3d4dd175204d53a247fc

**Comment by mcquin** [comment]
Interesting! I would expect the functions to product identical output.

There's not much we can do here, except to use `closing` in favor of `binary_closing`.

**Comment by 0x00b1** [comment]
@mcquin I was intrigued by this one too!

**Comment by karhohs** [comment]
I updated the [gist](https://gist.github.com/karhohs/bad263764e0b3d4dd175204d53a247fc) to show that all binary operations have this border effect, which differs from the non-binary morphology operations.

**Comment by karhohs** [comment]
FWIW, the MATLAB command *imerode* treats uint arrays and logical the same, i.e. there is no erosion of the edge of the image introducing a border.

**Comment by mcquin** [comment]
Closed #2839
